src/lightning_app/core/app.py

```diff
@@ -292,43 +292,39 @@ def _collect_deltas_from_ui_and_work_queues(self) -> t.List[t.Union[Delta, APIRe
         # during the `state_accumulate_wait` time.
         # The while loop can exit sooner if both queues are empty.
 
-        should_get_delta_from_api = True
-        should_get_component_output = True
+        keep_fetching = True
         deltas = []
         t0 = time()
 
         while (time() - t0) < self.state_accumulate_wait:
 
-            if self.api_delta_queue and should_get_delta_from_api:
-                delta_from_api: t.Union[DeltaRequest, APIRequest, CommandRequest] = self.get_state_changed_from_queue(
-                    self.api_delta_queue
-                )  # TODO: rename
-                if delta_from_api:
-                    if isinstance(delta_from_api, DeltaRequest):
-                        delta_from_api = delta_from_api.delta
-                    deltas.append(delta_from_api)
+            if self.delta_queue and keep_fetching:
+                # TODO: Fetch all available deltas at once to reduce queue calls.
+                delta: t.Optional[
+                    t.Union[DeltaRequest, APIRequest, CommandRequest, ComponentDelta]
+                ] = self.get_state_changed_from_queue(self.delta_queue)
```

Comment:
Here we are calling get from redis repeatedly: can’t we just call/expose `LRANGE` on redis so we can directly get all deltas currently on the queue in one call?
@hhsecond

Reply:
That would break the semantics of queue's `push`/`pop` behaviour but as far as we know that's the contract, it should be alright!

Reply:
but probably tricky because it's two different data structures - go fetching an array of python picked binaries (go arrays) and then client is python. We could introduce some common interface in between, like protobuf, perhaps

Reply:
Yes, this would be the best.

Reply:
Ok after a quick huddle we decided to create an extra endpoint and send back the whole queue as a JSON payload of  Base64 encoded deltas.

Reply:
Yes, let's do this in another PR as this is related to adding a new method to the base abstraction.